connect_four.py

```diff
@@ -25,48 +29,53 @@ def print_board(self) -> None:
 
             print(print_row)
             row_count -= 1
-        
-        print("  A  B  C  D  E  F  G")  #TODO needs to flex to board size
+
+        print_row = " "
+        for lett in range(self.num_cols):
+            print_row += " " + chr(lett+65) + " "
+        print(print_row)
 
 
     #function for checking the validity of a chosen column. Checks if col exists and is not full. 
-    def check_col(self, col: str): 
-        while True:        
-            if col not in self.col_dict: 
-                print(player1 + ", that column does not exist. Please pick another column (A-G):")
+    def check_col(self, col: str) -> int: 
+        while True:   
+            col_i = ord(col) #using UNICODE of capital letters, ord("A") == 65     
+            
+            if col_i < 65 or col_i > 64 + self.num_cols: #max board size is 26 (A-Z)
+                print(player1 + ", that column does not exist. Please pick another column ():") #TODO change to show flexible board size
                 col = str(input()).capitalize()
                 continue
             
-            elif len(self.col_dict[col][0]) >= self.num_rows:
-                print(player1 + ", that column is full. Please pick another column (A-G):")
+            elif len(self.columns[col_i-65]) >= self.num_rows:
```

Comment:
Try to avoid "magic numbers":  check out this link: https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad

I think it's better to make it explicit what 65 refers to, or could even just replace it with ord('A') i believe.